### PR TITLE
Story/MON-12

### PR DESCRIPTION
**Pull request by aa-kashk**
Opening a new PR after merging changes with develop.

### CODE REVIEW

**Review by jdamerow — CHANGES_REQUESTED** [review]
please fix indentation (should be 4 or 6 (if you follow what's already used) spaces but not 8) and follow our style guide before I keep reviewing.

**Comment by aa-kashk** [comment]
hope this is okay.

File: monitor/src/main/java/edu/asu/diging/monitor/core/db/impl/AppDbConnection.java [hunk 1]
```diff
@@ -13,75 +13,120 @@
...
         if (app.getId() == null) {
             throw new UnstorableObjectException("App does not have an id.");
         }
-        em.persist(app);
+        if (app.getRecipients() != null)
+            for (INotificationRecipient recipient : app.getRecipients()) {
+                recipient.getApps().add((App) app);
+                em.merge(recipient);
+            }
+        else {
```
Comment (jdamerow) [inline monitor/src/main/java/edu/asu/diging/monitor/core/db/impl/AppDbConnection.java]: please follow our styleguide

File: monitor/src/main/java/edu/asu/diging/monitor/core/model/impl/App.java [hunk 2]
```diff
@@ -23,7 +27,22 @@
...
+	}
+	
+	@Override
+	public void setRecipients(List<NotificationRecipient> recipients) {
+		this.recipients = recipients;
+	}
+	@Override
+	public String toString() {
+		return name;
+	}
```
Comment (jdamerow) [inline monitor/src/main/java/edu/asu/diging/monitor/core/model/impl/App.java]: formatting

File: monitor/src/main/java/edu/asu/diging/monitor/core/service/impl/AppHelper.java [hunk 3]
```diff
@@ -1,23 +1,62 @@
...
+        app.setExpectedReturnCodes(appForm.getExpectedReturnCodes());
+        app.setHealthUrl(appForm.getHealthUrl());
+        app.setName(appForm.getName());
+        app.setRetries(appForm.getRetries());
+        app.setTimeout(appForm.getTimeout());
+        app.setWarningReturnCodes(appForm.getWarningReturnCodes());
+        app.setPingInterval(appForm.getPingInterval());
+        app.setMethod(appForm.getMethod());
+        if (appForm.getRecipientIds()!=null)
+            app.setRecipients(getRecipientsById(appForm.getRecipientIds()));
```
Comment (jdamerow) [inline monitor/src/main/java/edu/asu/diging/monitor/core/service/impl/AppHelper.java:33]: code style

File: monitor/src/main/java/edu/asu/diging/monitor/web/admin/ModifyAppController.java [hunk 4]
```diff
@@ -12,36 +14,41 @@
...
+                app.getRecipients().stream().map(x -> x.getEmail()).collect(Collectors.toList()));
+        return "admin/apps/show";
+    }
+
+    @RequestMapping(value = "/admin/apps/{id}/modify", method = RequestMethod.POST)
+    public String update(@ModelAttribute AppForm appForm, @PathVariable("id") String id,
+            RedirectAttributes redirectAttrs) {
+        IApp app = appManager.getApp(id);
+        if (app.getRecipients() != null)
+            appManager.deleteExistingRecipients(app);
```
Comment (jdamerow) [inline monitor/src/main/java/edu/asu/diging/monitor/web/admin/ModifyAppController.java:46]: code style

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Comment by jdamerow** [comment]
once the formatting issues are fixed, should be good to go.

**Review by jdamerow — APPROVED** [review]

**Pull request merged**